### PR TITLE
Allow base 4.15

### PR DESCRIPTION
**Pull request by felixonmars**
Builds fine and all tests pass.

### CODE REVIEW

**Comment by jneira** [comment]
This is needed for allow the haskell-language-server brittany plugin for ghc-9.0.1 in hackage.
@lspitzner any chance to get it merged and complete brittany support for ghc-9.0.1? thanks in advance!

File: multistate.cabal [hunk 1]
```diff
@@ -68,7 +68,7 @@ library {
   other-modules:
     Control.Monad.Trans.MultiGST.Common
   build-depends:
-    base         >= 4.11  && <4.15,
+    base         >= 4.11  && <4.16,
```
Comment (jneira) [inline multistate.cabal:71]: I guess the package will be buildable with 4.16 as well
```suggestion
    base         >= 4.11  && <4.17,
```

Comment (jneira) [reply multistate.cabal:71]: No it is not, several errors like

```
[ 7 of 22] Compiling Control.Monad.Trans.MultiWriter.Strict ( src\Control\Monad\Trans\MultiWriter\Strict.hs, dist\build\Control\Monad\Trans\MultiWriter\Strict.o )

src\Control\Monad\Trans\MultiWriter\Strict.hs:123:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad (MultiWriterT x m)’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
123 |   return = MultiWriterT . return
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by lspitzner** [comment]
Thanks! This was replaced by #10/#11 which bring support for up to `ghc-9.2`.

**Pull request closed without merging**